lib/model-mapper/entities/resource.js

```diff
@@ -0,0 +1,45 @@
+import { sparqlEscape } from 'mu';
+
+export class Resource {
+
+  constructor(key, uri, model) {
+    /**
+     * Internal key to identify the resource
+     */
+    this.key = key;
+    this.uri = uri;
+    this.model = model;
+    this.properties = [];
+  }
+
+  get model() {
+    return this._model;
+  }
+
+  set model(model) {
+    this._model = model;
+    this._model.resources[this.key] = this;
+  }
+
+  /**
+   * NOTE: Because the UUID and URI are usually linked, when setting the URI, the UUID is reset.
```

Comment:
The uuid and the uri might be linked but it's also valid if it's not, would it be a big change not to assume they will contain the same uuid part ?